students/GuyCChan/session04/trigrams.py

```diff
@@ -0,0 +1,82 @@
+import string
+import random
+
+def read_in_data():
+# Open text file and put file body in a list.
+    with open('./sherlock_small.txt', 'rb') as file:
+        file_body = file.readlines()
+        allwords = []
+        for line in file_body:
+            line = line.strip()
+            if line.endswith("in part my own."):
+                break
+            allwords.append(line)
+# I'm doing it the beginner's way. Definitely not Pythonic.
+        txt_words = ' '.join([str(word) for word in allwords])
+        print(type(txt_words))
+        # I struggled with this part for a long time, maybe a couple of days!.
+        #NEED HELP HERE.
+        # In line 16, txt_words is a str type. One line later, in line 21,
+        # txt is a list! Why?
+        txt = txt_words.lower().split(' ')
+        print(type(txt), "here")
+# Remove punctuations and make a long string of words from text.
+        txt = txt.replace('"','')
+        txt = txt.replace('--', ' ')
+        txt = txt.replace(',', '')
+        txt = txt.replace('\\n','')
+        txt = txt.replace('(', '')
+        txt = txt.replace(')', '')
+        txt = txt.replace('.', '')
+        txt = txt.replace('-', ' ')
+        txt = txt.replace('\'', '')
+        txt = txt.replace('[', '')
+        txt = txt.replace(']', '')
+        txt = txt.replace(' ', '')
+        txt = txt.replace(' i ', ' I ')
+        txt = txt.replace('holmes', 'Holmes')
+        sp = " "
+        full_txt = sp.join(txt)
+        print(full_txt)
+        return full_txt
+
+def build_trigrams(full_txt):
+# Make a dictionary using a for loop. This section is heavily Chris-"inspired".
+# ipython sometimes would say "full_txt" is not defined. Why?
```

Comment:
because the name needs to be assigned in order to call this function -- if you hadn't already run code that defined it, you'd get that error.